### ISSUE TITLE
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled

### ISSUE THREAD

**Issue by pedros89** [kurimawxx00/webota-esp32#1]
Hi,

I followed exactly your great tutorial and tested on two ESP32 boards. Your solutions would really sort out a problem I have as I need to update ESP32 that gets the .bin from my website.
Unfortunately I got this error. 
This is the error
`Guru Meditation Error: Core 1 panic'ed (LoadProhibited). Exception was unhandled`
It seems it is trying to write the buffer in restricted memory area and it gets blocked.

```
WiFi connected
IP address: 
192.168.1.137
Response: 301
FW Size: 303
Updating firmware...
...Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.
Core 1 register dump:
PC      : 0x4000c2e0  PS      : 0x00060030  A0      : 0x80087bfa  A1      : 0x3ffb1db0  
A2      : 0x3ffb1dd0  A3      : 0x00000000  A4      : 0x00000010  A5      : 0x3ffb1dd0  
A6      : 0x00060823  A7      : 0x00000001  A8      : 0x00000000  A9      : 0x00150000  
A10     : 0x3ffbda68  A11     : 0x3ffc1984  A12     : 0x3ffc1884  A13     : 0x3ffb1f8c  
A14     : 0x3ffb1e7c  A15     : 0x00000008  SAR     : 0x0000000b  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x00000000  LBEG    : 0x4000c2e0  LEND    : 0x4000c2f6  LCOUNT  : 0x00000000  

ELF file SHA256: 0000000000000000

Backtrace: 0x4000c2e0:0x3ffb1db0 0x40087bf7:0x3ffb1dc0 0x401590fd:0x3ffb1e10 0x400d41ce:0x3ffb1e30 0x400d3cca:0x3ffb1e50 0x400d3dfa:0x3ffb1e70 0x400d3fe2:0x3ffb1e90 0x400d0823:0x3ffb1ec0 0x400d09db:0x3ffb1ee0 0x400d5796:0x3ffb1fb0 0x40089a32:0x3ffb1fd0

Rebooting...

⸮⸮Ih⸮bڜM⸮⸮⸮⸮⸮⸮⸮.....
WiFi connected
IP address: 
192.168.1.137
Response: 301
FW Size: 303
Updating firmware...
...Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.
Core 1 register dump:
PC      : 0x4000c2e0  PS      : 0x00060030  A0      : 0x80087bfa  A1      : 0x3ffb1db0  
A2      : 0x3ffb1dd0  A3      : 0x00000000  A4      : 0x00000010  A5      : 0x3ffb1dd0  
A6      : 0x00060823  A7      : 0x00000001  A8      : 0x00000000  A9      : 0x00150000  
A10     : 0x3ffbda68  A11     : 0x3ffc1984  A12     : 0x3ffc1884  A13     : 0x3ffb1f8c  
A14     : 0x3ffb1e7c  A15     : 0x00000008  SAR     : 0x0000000b  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x00000000  LBEG    : 0x4000c2e0  LEND    : 0x4000c2f6  LCOUNT  : 0x00000000  

ELF file SHA256: 0000000000000000

Backtrace: 0x4000c2e0:0x3ffb1db0 0x40087bf7:0x3ffb1dc0 0x401590fd:0x3ffb1e10 0x400d41ce:0x3ffb1e30 0x400d3cca:0x3ffb1e50 0x400d3dfa:0x3ffb1e70 0x400d3fe2:0x3ffb1e90 0x400d0823:0x3ffb1ec0 0x400d09db:0x3ffb1ee0 0x400d5796:0x3ffb1fb0 0x40089a32:0x3ffb1fd0

Rebooting...
⸮z⸮⸮,⸮$I⸮m :
```

It is just carrying on like this forever.

These are my settings 
![image](https://user-images.githubusercontent.com/30262131/125298419-65243200-e328-11eb-971d-5ec1921565ac.png)

From the search I did maybe there is no final character \n   But I'm still looking for a solution
https://flashgamer.com/blog/comments/guru-meditation-error-core-0-paniced-loadprohibited

How did you make it work for yourself?

**Comment by rppelayo** [comment]
Hi,

Please use the ESP32 exception decoder tool to give you more info about that error. https://github.com/me-no-dev/EspExceptionDecoder

**Comment by pedros89** [comment]
Hi thank you for your reply. I will download the tool for debugging because it looks super useful, but finally after some time I have sorted it out.

I believe that the problem could be that I was connecting with  `"http://....."` to my server but my server has https enabled with a certificate.

What I did was to use this example provided in the update library that is especially made for https. It is very important to format the certificate string as in the example with  `\n`  at the end of every line

To all people that get the error I got, if you have an https server use the example at this link
https://github.com/espressif/arduino-esp32/tree/master/libraries/Update/examples/HTTPS_OTA_Update

**Comment by woutput** [comment]
Hi @pedros89 , I had the same issue but solved it:
In my case, the HTTP response was 301: Moved Permanently. I think that this is caused by the .htaccess file on my webserver. It was configured (by WordPress) to replace the unencrypted http:// by the encrypted https://
In my case the solution was simple: modify my URL (as #defined in HOST) to include the s in https:// so that no redirect was needed. After the change, the HTTP response was 200: OK
An improvement for the code that is hosted here would be to either: only accept HTTP response 200 or (even better) follow the 301 redirect.
I hope this helps.

**Comment by rppelayo** [comment]
Will mark this issue as resolved

**Comment by pedros89** [comment]
Hello @woutput, thank you for pointing that out. In my case because of that error I had abandoned this webota library, in addition I needed to have https, only http was not secure enough for my application.
I have developed a repository that allows you to update both ESP App and ESP spiffs and this is the one I am using at the moment. I am finding that the code based on [the official ESP32 idf OTA](https://github.com/espressif/esp-idf/blob/master/examples/system/ota/native_ota_example/main/native_ota_example.c) is very reliable. Feel free to look at this repository:
https://github.com/pedros89/ESP32-update-both-OTA-Firmware-and-OTA-SPIFFS-with-Arduino
